docs/usage/sat_bootprep.md

```diff
@@ -344,6 +344,17 @@ values can be viewed by running `sat bootprep list-vars`. For more information
 on options that may be used with the `list-vars` subcommand, refer to the man page
 for the `sat bootprep` subcommand.
 
+#### Using Hyphens in HPC CSM Software Recipe Variables
+
+Variable names with hyphens are not allowed in Jinja2 expressions, because they
+will be parsed as an arithmetic expression rather than a single variable.
+In order to support product names with hyphens, `sat bootprep` will convert
+hyphens to underscores in all top-level keys in the software recipe variables,
+as well as any variables passed to the command using `--vars` or `--vars-file`.
+For example, to refer to the version of the product
+`slingshot-host-software` in the bootprep input file, you would write
+`"{{slingshot_host_software.version}}"`.
```

Comment:
```suggestion
Variable names with hyphens are not allowed in Jinja2 expressions because they
are parsed as an arithmetic expression instead of a single variable.
To support product names with hyphens, `sat bootprep` converts
hyphens to underscores in all top-level keys in the software recipe variables.
It also converts any variables passed to the command using `--vars` or `--vars-file`.
For example, to refer to the version of the product
`slingshot-host-software` in the bootprep input file, you would write
`"{{slingshot_host_software.version}}"`.
```

Reply:
Suggestion looks good to me, but worth working out the conversation in https://github.com/Cray-HPE/docs-sat/pull/70#discussion_r1092399837